source/MaterialXShaderGen/ShaderGenerators/Glsl/LightGlsl.cpp

```diff
@@ -0,0 +1,112 @@
+#include <MaterialXShaderGen/ShaderGenerators/Glsl/LightGlsl.h>
+#include <MaterialXShaderGen/Util.h>
+
+namespace MaterialX
+{
+
+SgImplementationPtr LightGlsl::creator()
+{
+    return std::make_shared<LightGlsl>();
+}
+
+void LightGlsl::initialize(ElementPtr implementation, ShaderGenerator& shadergen)
+{
+    GlslImplementation::initialize(implementation, shadergen);
+
+    ImplementationPtr impl = implementation->asA<Implementation>();
+    if (!impl)
+    {
+        throw ExceptionShaderGenError("Element '" + implementation->getName() + "' is not an Implementation element");
+    }
+
+    const string& file = impl->getAttribute("file");
+    if (file.empty())
+    {
+        throw ExceptionShaderGenError("No source file specified for implementation '" + impl->getName() + "'");
+    }
+
+    // Find the function name to use
+    _functionName = impl->getAttribute("function");
+    if (_functionName.empty())
+    {
+        // No function given so use nodedef name
+        _functionName = impl->getNodeDefString();
+    }
+
+    _functionSource = "";
+    if (!readFile(shadergen.findSourceCode(file), _functionSource))
+    {
+        throw ExceptionShaderGenError("Can't find source file '" + file + "' used by implementation '" + impl->getName() + "'");
+    }
+
+    NodeDefPtr nodeDef = impl->getNodeDef();
+    _lightUniforms.resize(nodeDef->getInputCount() + nodeDef->getParameterCount());
+
```

Comment:
Extract all inputs/parameters on the light shader node. These turn into members in the LightData uniform block later.